app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,51 @@
+class OrdersController < ApplicationController
+  before_action :get_user, only: [:index, :new, :create]
+  before_action :get_cart_and_sum, only: [:new, :create]
+
+  def index
+    @orders = Order.where(user_id: @user_id) 
+  end
+  
+  def new
+    @order = Order.new
+  end
+
+  def show
+    @order = Order.find(params[:id])
+    @cart = @order.cart_items
+    @sum = 0
+    @cart.each do |ci| 
+      @sum += ci.price_on_purchase * ci.quantity
+    end
```

Comment:
```suggestion
    @cart.each do |cart_item| 
      @sum += cart_item.price_on_purchase * cart_item.quantity
    end
```
acho mais legivel dessa forma, vou deixar como sugestão